### ISSUE TITLE
Multiple Definitions of `cmd_argv' and `cmd_argc'

### ISSUE THREAD

**Issue by charJe** [part-cw/lambdanative#294]
I'm just trying to build the calculator app, but I get the same error when trying to build any app. 
I get these two errors:
```
/usr/bin/ld: /home/charles/.cache/lambdanative/linux/lib/libpayload.a(system.o):(.bss+0x0): multiple definition of `cmd_argv'; /home/charles/.cache/lambdanative/linux/lib/libpayload.a(2978470258.o):(.bss+0x0): first defined here
/usr/bin/ld: /home/charles/.cache/lambdanative/linux/lib/libpayload.a(system.o):(.bss+0x8): multiple definition of `cmd_argc'; /home/charles/.cache/lambdanative/linux/lib/libpayload.a(2978470258.o):(.bss+0x8): first defined here
collect2: error: ld returned 1 exit status
ERROR: failed with exit code 1
BUILD FAILED
make: *** [Makefile:2: all] Error 1
```
OS: Fedora 32
GCC: 10.1.1
Gambit-c: 4.9.3

**Comment by zwieblum** [comment]
Works here (gcc version 8.3.0 + Devuan Beowulf)

Is this the first build on linux? When things go wrong for me I usually do "make clean; make scrub; rm ~/.lambdanatve ~/.cache/lambdanate -R"

**Comment by charJe** [comment]
It is my first time. Even after trying your commands, the error persists.

**Comment by mgorges** [comment]
There is something broken in your build system - can you configure in verbose mode `./configure DemoConsole debug verbose; make` and post the full log as a https://gist.github.com ?

**Comment by charJe** [comment]
https://gist.github.com/charJe/a175f9e68d12d4b1732afb273d8d5628

**Comment by charJe** [comment]
Would it be possible to use the `--allow-multiple-definition` option for `/usr/bin/ld`?

**Comment by mgorges** [comment]
Sure you could try setting that in `targets/linux/host_linux`, but there is an issue in your system causing multiple includes, which might be better to resolve instead? Also, you are configured to build 'pngtool' not a a LambdaNative app?

**Comment by charJe** [comment]
I used `./configure DemoConsole debug verbose` after clearing any caches.

**Comment by charJe** [comment]
Even after changing `SYS_LD="ld --allow-multiple-definition"` in `targets/linux/host_linux`, the same error persists.

**Comment by charJe** [comment]
> Also, you are configured to build 'pngtool' not a a LambdaNative app?
It appears to me that the 'pngtool' is required for the building of any other app such as Calculator or DemoConsole. The building of the pngtool is what is failing. 

In `pngtool/main.scm` I found functions `system-cmdargv` and `system-cmdargc` which seem to basically be defined as C functions in both `libraries/liblambdanative/system.c` and `libraries/liblambdanative/lambdanative.h` through the c-lambdas in `modules/config/config.scm`. I'm not sure how to fix it though.

**Comment by charJe** [comment]
> Also, you are configured to build 'pngtool' not a a LambdaNative app?

It looks to me that 'pngtool' is required for other apps and building it is what my actual issue is.

**Comment by mgorges** [comment]
Yes, a few tools are required to be build and the configuration script updates itself in terms of targets if they haven't been completed yet to make sure that happens. As such `pngtool` is simply a special `LambdaNative` console app. Why you get a conflict with duplicate definitions of the standard C-functions is still not clear to me though.

**Comment by charJe** [comment]
I can get it to work if I _always_ extern the variables in `libraries/liblambdanative/`
```c
/* #ifdef USECONSOLE */
extern 
/* #endif */
char **cmd_argv;

/* #ifdef USECONSOLE */
extern 
/* #endif */
int cmd_argc;
```

**Comment by charJe** [comment]
Interestingly enough, these two variables must be conditionally `extern`ed to compile for android.

**Comment by mgorges** [comment]
`USECONSOLE` is set in [targets/linux/build-binary#L60](https://github.com/part-cw/lambdanative/blob/master/targets/linux/build-binary#L60) as a macro to be used by the GCC preprocessor with `-DUSECONSOLE` as such I don't understand why this could be the case on your system.

**Comment by MadEarl** [comment]
Can confirm problem for Fedora 32, also with gcc 10.2.1. Workaround as described by charJe works here, too. Comment #ifdef for linux build, remove comment for android build.

**Comment by goober99** [comment]
Can also confirm problem for Debian Sid with GCC 10.2.0.

**Comment by mgorges** [comment]
Should be closed in b803c2b as was issue #410? If not, please reopen.
